src/bg_cache_worker.ts

```diff
@@ -0,0 +1,151 @@
+
+      // If voter not already added as a member on the DAO set them up (this should not happen right?)
+      if (!initialState.daos[proposalDetails.daoAvatarAddress].members[voteEventArgs._voter]) {
+        initialState.daos[proposalDetails.daoAvatarAddress].members[voteEventArgs._voter] = { reputation: 0, tokens: 0, votes: {}, stakes: {}};
+      }
+      initialState.daos[proposalDetails.daoAvatarAddress].members[voteEventArgs._voter].votes[voteEventArgs._proposalId] = {
+        avatarAddress: proposalDetails.daoAvatarAddress,
+        proposalId: voteEventArgs._proposalId,
+        reputation: Util.fromWei(voteEventArgs._reputation),
+        transactionState: TransactionStates.Confirmed,
+        vote: Number(voteEventArgs._vote),
+        voterAddress: voteEventArgs._voter
+      }
+    }
+  } catch (e) {
+    console.error("Error getting votes: ", e)
+  }
+
+  try {
+    // Get all stakes ever made
+    const stakeEventWatcher = genesisProtocol.Stake({}, { fromBlock: lastBlock });
+    const getStakeEvents = promisify(stakeEventWatcher.get.bind(stakeEventWatcher));
+    const stakeEvents = await getStakeEvents();
+    let stakeEventArgs;
+    for (let cnt = 0; cnt < stakeEvents.length; cnt++) {
+      stakeEventArgs = stakeEvents[cnt].args;
+      proposalDetails = initialState.proposals[stakeEventArgs._proposalId];
+
+      // If staker not already added as a member on the DAO set them up (this should not happen right?)
+      if (!initialState.daos[proposalDetails.daoAvatarAddress].members[stakeEventArgs._voter]) {
+        initialState.daos[proposalDetails.daoAvatarAddress].members[stakeEventArgs._voter] = { reputation: 0, tokens: 0, votes: {}, stakes: {}};
+      }
+      initialState.daos[proposalDetails.daoAvatarAddress].members[stakeEventArgs._voter].stakes[stakeEventArgs._proposalId] = {
+        avatarAddress: proposalDetails.daoAvatarAddress,
+        proposalId: stakeEventArgs._proposalId,
+        transactionState: TransactionStates.Confirmed,
+        stake: Util.fromWei(stakeEventArgs._amount),
+        prediction: Number(stakeEventArgs._vote),
+        stakerAddress: stakeEventArgs._voter
+      }
+    }
+  } catch (e) {
+    console.error("Error getting stakes: ", e);
+  }
+
+  // Write cached blockchain data to S3
+  aws.config.region = 'us-west-2';
+  const s3 = new aws.S3();
+  const fileName = 'initialArcState-' + network + '.json';
+  const fileType = 'application/json';
+  const s3Params = {
+    Body: JSON.stringify(initialState),
+    Bucket: process.env.S3_BUCKET || 'daostack-alchemy',
+    Key: fileName,
+    //Expires: 60,
+    ContentType: fileType,
+    ACL: 'public-read'
+  };
+  s3.putObject(s3Params, (err, data) => {
+    if (err) {
+      console.error("Error writing data to S3 = ", err, err.stack);
+    } else {
+      // tslint:disable-next-line:no-console
+      console.log("Successfully wrote cached data for " + network + " to S3. ", data);
+    }
+  });
+
+  setTimeout(cacheBlockchain, 1000);
```

Comment:
How long does the function take to execute?  Just wondering if one second is a good value.  (particularly since the function is still caching the entire blockchain)

Reply:
maybe 30-40 seconds? basically I just restart it when it ends :)

Reply:
Shouldn't the script create a redux `store` and dispatch actions to the already existing reducers to update it (eg. `STAKE_FULFILLED`, `VOTE_FULFILLED`)?